okhttp-doh/src/main/java/okhttp3/doh/DnsOverHttps.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2014 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.doh;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Map;
+import okhttp3.Dns;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.internal.platform.Platform;
+import okio.ByteString;
+
+// TODO implement rules and caching according to spec
+public class DnsOverHttps implements Dns {
+  private final OkHttpClient client;
+  private final String urlPrefix;
+  private final Map<String, List<InetAddress>> bootstrapAddresses;
```

Comment:
This is interesting!

Reply:
Replaced this with a Dns instance, Dns.SYSTEM would bootstrap normally.  But still allowing to bootstrap with your well known providers and avoid any dependency on normal Dns infrastructure.